pennylane/drawer/tape_mpl.py

```diff
@@ -119,13 +122,19 @@ def _tape_mpl(tape, wire_order=None, show_all_wires=False, decimals=None, **kwar
         for op in layer_ops:
             specialfunc = special_cases.get(op.__class__, None)
             if specialfunc is not None:
-                mapped_wires = [wire_map[w] for w in op.wires]
+                mapped_wires = (
+                    [wire_map[w] for w in op.wires] if len(op.wires) != 0 else wire_map.values()
+                )
                 specialfunc(drawer, layer, mapped_wires, op)
 
             else:
                 op_control_wires = getattr(op, "control_wires", [])
                 control_wires = [wire_map[w] for w in op_control_wires]
-                target_wires = [wire_map[w] for w in op.wires if w not in op_control_wires]
+                target_wires = (
```

Comment:
I guess control SnapShot does not exist haha Is there an example to test this change?

Reply:
If an operation had control wires then `len(op.wires)` cannot be zero. Testing whether `qml.Snapshot` works correctly implicitly tests this, but I'm not sure how else to test it as any controlled operation will always have `op.wires` be non-empty, and cases where someone tries to initialize an operation with non-zero control wires but zero target wires should always fail, and testing for that shouldn't be in the `drawer` module.